src/serverenvironment.cpp

```diff
@@ -2068,6 +2069,7 @@ void ServerEnvironment::deactivateFarObjects(bool _force_delete)
 				force_delete = true;
 		}
 
+		obj->markForDeactivation();
```

Comment:
Let's try an experiment: Deactivate things in the function that was made for deactivating things. (wow!)
Please test.

I don't know what side effects this might bring about because the object lifecycle is just a big, big mess, but it seems to work.

Reply:
That does not sound very convincing, but the location seems right. However, this will need a comment like
`// Trigger the script callback after saving and prior deletion`

Will test.

Reply:
Well I tried this and it worked, and it makes sense to deactivate things both when you delete and when you unload them.
The alternative was a monstrous condition along the lines of: _check if the object has been not really deactivated but in need of a deactivate call, and call the deactivate event through a special code path even thought we're not really deactivating._ That doesn't sound quite right. In other words, I'd have to add a third deactivation path, even though two are already too many.

I'm not sure what kind of comment could express the intent here properly, because this line of code has quite a history now:
- It was a bool assignment that among other things made isGone evaluate to true
- It got converted to a call that also tripped the deactivation, because deactivation should always happen before isGone is true
- There was still a code path where neither deactivation path would be called
- The line was moved out of its block to execute unconditionally because that actually makes a lot of sense in its new context

I'll think of something.

Reply:
Done, let me know if this is enough.